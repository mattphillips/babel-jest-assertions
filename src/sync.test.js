const pluginTester = require('babel-plugin-tester');
const plugin = require('./');

pluginTester({
  plugin,
  tests: {
    'Does not modify code when not a test': {
      code: `
      const add = (a, b) => a + b;
      `
    },
    'Adds number of assertions and has assertions check when given one expect statement': {
      snapshot: true,
      code: `
      describe('.add', () => {
        it('returns 1 when given 0 and 1', () => {
          expect(add(0, 1)).toEqual(1);
        });
      });
      `
    },
    'Adds total number of assertions when given multiple expect statements': {
      snapshot: true,
      code: `
      describe('.add', () => {
        it('returns 1 when given 0 and 1', () => {
          expect(add(0, 1)).toEqual(1);
          expect(add(1, 0)).toEqual(1);
        });
      });
      `
    },
    'Adds total number of assertions when given multiple expect statements in nested if scope': {
      snapshot: true,
      code: `
      describe('.add', () => {
        it('returns 1 when given 0 and 1', () => {
          if (true) {
            expect(add(0, 1)).toEqual(1);
          } else if (false) {
            expect(add(1, 0)).toEqual(1);
          }
        });
      });
      `
    },
    'Does not add expect.assertions when expect.assertions is supplied': {
      snapshot: true,
      code: `
      describe('.add', () => {
        it('returns 1 when given 0 and 1', () => {
          expect.assertions(1);
          expect(add(0, 1)).toEqual(1);
        });
      });
      `
    },
    'Adds expect.assertions when expect.hasAssertions is supplied': {
      snapshot: true,
      code: `
      describe('.add', () => {
        it('returns 1 when given 0 and 1', () => {
          expect.hasAssertions();
          expect(add(0, 1)).toEqual(1);
        });
      });
      `
    },
    'Does not count commented out expect statements': {
      snapshot: true,
      code: `
      describe('.add', () => {
        it('returns 1 when given 0 and 1', () => {
          // expect(add(1, 2)).toEqual(3);
          expect(add(0, 1)).toEqual(1);

          /*
            expect(add(4, 5).toEqual(9));
          */

          const a = 1; // expect(a).toEqual(1);
          const b = 2; /* expect(b).toEqual(2); */
          
          expect(add(1, 0)).toEqual(1);
          /*
            expect(add(6, 1).toEqual(7));
            */
        });
      });
      `
    }
  }
});
