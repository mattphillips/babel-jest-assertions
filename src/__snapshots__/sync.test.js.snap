// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds expect.assertions when expect.hasAssertions is supplied 1`] = `
"
describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect.hasAssertions();
    expect(add(0, 1)).toEqual(1);
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect.assertions(1);

    expect.hasAssertions();
    expect(add(0, 1)).toEqual(1);
  });
});
"
`;

exports[`Adds number of assertions and has assertions check when given one expect statement 1`] = `
"
describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect(add(0, 1)).toEqual(1);
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect.assertions(1);
    expect.hasAssertions();

    expect(add(0, 1)).toEqual(1);
  });
});
"
`;

exports[`Adds total number of assertions when given multiple expect statements 1`] = `
"
describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect(add(0, 1)).toEqual(1);
    expect(add(1, 0)).toEqual(1);
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect.assertions(2);
    expect.hasAssertions();

    expect(add(0, 1)).toEqual(1);
    expect(add(1, 0)).toEqual(1);
  });
});
"
`;

exports[`Adds total number of assertions when given multiple expect statements in nested if scope 1`] = `
"
describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    if (true) {
      expect(add(0, 1)).toEqual(1);
    } else if (false) {
      expect(add(1, 0)).toEqual(1);
    }
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect.assertions(2);
    expect.hasAssertions();

    if (true) {
      expect(add(0, 1)).toEqual(1);
    } else if (false) {
      expect(add(1, 0)).toEqual(1);
    }
  });
});
"
`;

exports[`Does not add expect.assertions when expect.assertions is supplied 1`] = `
"
describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect.assertions(1);
    expect(add(0, 1)).toEqual(1);
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect.hasAssertions();

    expect.assertions(1);
    expect(add(0, 1)).toEqual(1);
  });
});
"
`;

exports[`Does not count commented out expect statements 1`] = `
"
describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    // expect(add(1, 2)).toEqual(3);
    expect(add(0, 1)).toEqual(1);

    /*
      expect(add(4, 5).toEqual(9));
    */

    const a = 1; // expect(a).toEqual(1);
    const b = 2; /* expect(b).toEqual(2); */
    
    expect(add(1, 0)).toEqual(1);
    /*
      expect(add(6, 1).toEqual(7));
      */
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

describe('.add', () => {
  it('returns 1 when given 0 and 1', () => {
    expect.assertions(2);
    expect.hasAssertions();

    // expect(add(1, 2)).toEqual(3);
    expect(add(0, 1)).toEqual(1);

    /*
      expect(add(4, 5).toEqual(9));
    */

    const a = 1; // expect(a).toEqual(1);
    const b = 2; /* expect(b).toEqual(2); */

    expect(add(1, 0)).toEqual(1);
    /*
      expect(add(6, 1).toEqual(7));
      */
  });
});
"
`;
